Userland/Applications/TextEditor/main.cpp

```diff
@@ -30,6 +31,50 @@ int main(int argc, char** argv)
 
     parser.parse(argc, argv);
 
+    // Unveiling here so we can take advantage of the input file name
+    if (unveil("/home/anon/.config", "rw") < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    if (unveil("/res", "r") < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    if (unveil("/tmp/portal/webcontent", "rw") < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    if (unveil("/tmp/portal/fileaccess", "rw") < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    if (file_to_edit) {
+        FileArgument parsed_argument(file_to_edit);
+
+        if (!parsed_argument.filename().starts_with("/"sv)) {
+            char* buf = getcwd(nullptr, 0);
+            LexicalPath lexical_path = LexicalPath::join(String(buf), parsed_argument.filename());
```

Comment:
Same as above with the unnecessary explicit `String(buf)` and `auto lexical_path`.